File: app/src/main/java/com/karthik/splash/SplashApp.java
package com.karthik.splash;

import android.app.Application;

import com.karthik.splash.Components.DaggerSplashAppComponent;
import com.karthik.splash.Components.SplashAppComponent;
import com.karthik.splash.Modules.SplashAppModule;

/**
 * Created by karthikrk on 12/11/17.
 */

public class SplashApp extends Application{

    private SplashAppComponent splashAppComponent;


    @Override
    public void onCreate() {
        super.onCreate();
        splashAppComponent = DaggerSplashAppComponent.builder()
                .splashAppModule(new SplashAppModule(this))
                .build();
        splashAppComponent.inject(this);
    }


    public SplashAppComponent getComponent(){
        return splashAppComponent;
    }
}

File: app/src/main/java/com/karthik/splash/root/SplashApp.kt
package com.karthik.splash.root

import android.app.Application
import androidx.room.Room
import com.karthik.splash.storage.db.SplashDatabase

class SplashApp : Application() {
    private lateinit var splashAppComponent: SplashAppComponent

    override fun onCreate() {
        super.onCreate()
        val splashDatabase = Room.databaseBuilder(this,SplashDatabase::class.java,"splashdb").build()
        splashAppComponent = DaggerSplashAppComponent.builder()
                .splashAppModule(SplashAppModule(this,splashDatabase))
                .build()
        splashAppComponent.inject(this)
    }

    fun getComponent() = splashAppComponent
}